addons/website_sale/models/product_ribbon.py

```diff
@@ -16,8 +20,39 @@ class ProductRibbon(models.Model):
         required=True,
         default='left',
     )
+    style = fields.Selection(
+        string="Style",
+        selection=[('ribbon', "Ribbon"), ('tag', "Badge")],
+        required=True,
+        default='ribbon',
+    )
+    assign = fields.Selection(
+        string="Assign",
+        selection=[
+            ('manual', "Manually"),
+            ('sale', "Sale"),
+            ('new', "New"),
+        ],
+        required=True,
+        default='manual',
+    )
+    new_period = fields.Integer(default=30)
+
+    @api.constrains('assign')
+    def _check_assign(self):
+        for ribbon in self:
+            if ribbon.assign != 'manual':
+                existing_ribbons = self.search([
+                    ('id', '!=', ribbon.id),
+                    ('assign', '=', ribbon.assign)
+                ], limit=1)
+                if existing_ribbons:
+                    raise ValidationError(
+                        _(
+                            "Only one ribbon with the assign %s is allowed.",
+                            dict(self._fields['assign'].selection).get(ribbon.assign)
+                        )
+                    )
 
-    def _get_position_class(self):
-        if not self:
-            return 'd-none'
-        return 'o_ribbon_left' if self.position == 'left' else 'o_ribbon_right'
+    def _get_css_classes(self):
+        return f'o_{self.style or "ribbon"} o_{self.position or "left"}'
```

Comment:
Please avoid computing class name strings on the fly. I make it difficult to look for those classes in Odoo later. Here is a suggestion that would allow to search the class name in Odoo:
```suggestion
    def _get_css_classes(self):
        css_classes = "o_ribbons"
        match self.style:
            case "left":
                 css_classes += " o_left"
            [...]
        return css_classes
```